Sources/BraveWallet/Crypto/Stores/PortfolioStore.swift

```diff
@@ -357,7 +357,17 @@ public class PortfolioStore: ObservableObject {
           TokenNetworkAccounts(
             token: token,
             network: networkAssets.network,
-            accounts: selectedAccounts.filter { $0.coin == token.coin }
+            accounts: selectedAccounts.filter {
+              if token.coin == .fil {
+                if token.chainId == BraveWallet.FilecoinMainnet {
+                  return $0.keyringId == BraveWallet.KeyringId.filecoin
+                } else {
+                  return $0.keyringId == BraveWallet.KeyringId.filecoinTestnet
+                }
```

Comment:
I believe you can use the new `BraveWallet.KeyringId.keyringId(for:_:)` extension here?
```suggestion
                return $0.keyringId == BraveWallet.KeyringId.keyringId(for: token.coin, token.chainId)
```